05/app.js

```diff
@@ -11,30 +11,18 @@ Student.prototype.addGrade = function (subject, grade) {
     this.grades[subject].push(grade);
 };
 
-function averageCounter(item){
-    let sum = 0;
-    let count = 0;
-    item.forEach(function (num) {
-        sum += num;
-        count++;
-    });
-    return sum / count;
-}
+Student.prototype.calculateAverage = function (gradesArray) {
+    const sum = gradesArray.reduce((acc, grade) => acc + grade, 0);
+    return sum / gradesArray.length;
+};
 
 Student.prototype.getAverageGrade = function (subject) {
     if (subject) {
-        return averageCounter(this.grades[subject])
+        return Number(this.calculateAverage(this.grades[subject]).toFixed(2));
```

Comment:
Warto sprawdzić czy `this.grades[subject]` nie jest `undefined` czyli czy istnieje taki przedmiot z ocenami